eldritch/test_mythos.py

```diff
@@ -2055,6 +2116,374 @@ def testNormal(self):
+    self.state.mythos.append(Mythos5())
+    self.state.event_stack.append(Mythos(self.char))
+    self.resolve_until_done()
+    self.assertEqual(self.monster.place.name, "Southside")
+    self.assertEqual(self.monster.movement(self.state), "normal")
+
+
+class Mythos42Test(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.state.turn_number = 0
+    self.state.turn_phase = "mythos"
+    self.mythos = Mythos42()
+    self.state.mythos.append(self.mythos)
+    self.state.event_stack.append(Mythos(self.char))
+    self.resolve_until_done()
+    self.assertEqual(self.mythos, self.state.environment)
+    self.assertNotIn(self.mythos, self.state.mythos)
+
+  def testCostIsZero(self):
+    self.char.possessions.append(items.Voice(0))
+    self.char.possessions.append(items.DreadCurse(0))
+    self.advance_turn(0, "upkeep")
+    self.assertEqual(self.char.sanity, 5)
+    voice = self.resolve_to_usable(0, "Voice0")
+    self.state.event_stack.append(voice)
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      cast_choice = self.resolve_to_choice(MultipleChoice)
+      cast_choice.resolve(self.state, "Cast")
+      sliders = self.resolve_to_choice(SliderInput)
+      sliders.resolve(self.state, "done", None)
+      self.resolve_until_done()
+      self.assertEqual(self.char.sanity, 5)
+
+      self.state.event_stack.append(Combat(self.char, monsters.FormlessSpawn()))
+      fight_evade = self.resolve_to_choice(FightOrEvadeChoice)
+      fight_evade.resolve(self.state, "Fight")
+      curse = self.resolve_to_usable(0, "Dread Curse0")
+      self.state.event_stack.append(curse)
+      cast_choice = self.resolve_to_choice(MultipleChoice)
+      cast_choice.resolve(self.state, "Cast")
+      weapons = self.resolve_to_choice(CombatChoice)
+      weapons.resolve(self.state, "done")
+      self.resolve_until_done()
+      self.assertEqual(self.char.sanity, 5)
+
+  def testEnvironmentTakesPrecedence(self):
+    # FAQ p. 10
+    class Blight(GlobalEffect):
+      def __init__(self):
+        self.name = "TestEffect"
+
+      def get_modifier(self, thing, attribute, state):
+        if isinstance(thing, items.Spell) and attribute == "sanity_cost":
+          return 1
+        return 0
+
+    self.state.other_globals.append(Blight())
+    self.testCostIsZero()
+
+
+class Mythos51Test(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.state.turn_number = 0
+    self.state.turn_phase = "mythos"
+    self.mythos = Mythos51()
+    self.state.mythos.append(self.mythos)
+    self.state.event_stack.append(Mythos(self.char))
+    self.resolve_until_done()
+    self.assertEqual(self.mythos, self.state.environment)
+    self.assertNotIn(self.mythos, self.state.mythos)
+
+  def testCannotUse(self):
+    self.char.possessions.append(items.Voice(0))
+    self.advance_turn(0, "upkeep")
+    self.assertEqual(self.char.sanity, 5)
+    sliders = self.resolve_to_choice(SliderInput)
```

Comment:
Once you've reached the SliderInput, do you need to test that there is nothing in state.usables?

Reply:
I guess I didn't think I needed to explicitly assert that usables is empty, but it doesn't hurt.  Also adding a DreadCurse to your inventory so the check in combat asserts that it's not usable there.

Reply:
The reason to assert on usables being empty is that if the usables interrupt a choice event, they do not otherwise interrupt the flow. So if you resolve the choice event, you won't know whether anything was usable at the time of the choice.